api/v1alpha3/azurecluster_webhook.go

```diff
@@ -29,3 +35,40 @@ func (r *AzureCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		For(r).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-azurecluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azurecluster,versions=v1alpha3,name=validation.azurecluster.infrastructure.cluster.x-k8s.io
+
+var _ webhook.Validator = &AzureCluster{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *AzureCluster) ValidateCreate() error {
+	machinelog.Info("validate create", "name", r.Name)
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *AzureCluster) ValidateUpdate(old runtime.Object) error {
+	machinelog.Info("validate update", "name", r.Name)
+	azureCluster, ok := old.(*AzureCluster)
+	if !ok {
+		return fmt.Errorf("update object is not a AzureCluster type")
+	}
+
+	if azureCluster.Spec.Location != r.Spec.Location {
```

Comment:
@CecileRobertMichon @serbrech do we really want to create a re-usable function to validate that two strings aren't equal?

Reply:
I think @serbrech's point is that the _webhook file shouldn't contain the actual logic for validation but instead each object should have its own _validation file with functions that can each be unit tested separately. Right now it might be just a string comparison because you only setup the initial skeleton for Cluster update validation but as we add more validations the logic will become more complex. Might as well start things the right way so others can be guided to add further validations in the right place. See how `ValidateCronJob` is implemented [here](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html) for an example.

@serbrech please correct me if I'm wrong.

Reply:
+1 to what Cecile said, just to setup the structure for future validations